src/wfs/src/main/java/org/geoserver/wfs/WFSGetFeatureOutputFormat.java

```diff
@@ -202,11 +202,21 @@ public void write(Object value, OutputStream output, Operation operation)
 
     private <T> T getFeatureTypeInfoProperty(
             Catalog catalog, FeatureCollection features, Function<FeatureTypeInfo, T> callback) {
-        FeatureType featureType = features.getSchema();
-
-        ResourceInfo meta = catalog.getResourceByName(featureType.getName(), ResourceInfo.class);
+        FeatureTypeInfo fti;
```

Comment:
Is this change actually related with this PR associated issue ? I'm in doubt because of this comment ``// fix for complex features bad type name``.

Reply:
It is related, when FeatureCollection is instance of TypeInfoCollectionWrapper.Complex its getSchema() methos returns a "nameType" named FeatureType (instead "name" only). Maybe it deserves a better comment, changing it.  Thanks!